js/w3c/headers.js

```diff
@@ -155,6 +156,18 @@ define(
                     ret += ", <span class='ed_mailto'><a class='u-email email' " + rm + " href='mailto:" + p.mailto + "'>" + p.mailto + "</a></span>";
                 }
                 if (p.note) ret += " (" + p.note + ")";
+                if (p.extras && p.extras.length !== 0) {
```

Comment:
I hope you had great vacations.

As for the rewrite: as I said, I also prefer what you call declarative (I call it rather functional, but that is just a name) style. But I do not know how long it will take to get to the v4 and, in the meantime, it may not be good to have a mix of styles in the same file on such close functionalities... But I let you decide.